core/src/main/java/hudson/model/Slave.java

```diff
@@ -155,9 +157,10 @@
     private transient volatile Set<Label> labels;
 
     /**
-     * Id of user which creates this agent {@link User}.
+     * Removed with no replacement.
```

Comment:
Should there be more detail in the javadoc that explains the context why it was removed?  For example, should this comment include a link to JENKINS-21837 or possibly the majority of the text from the pull request?
```suggestion
     * Removed with no replacement.
     * This broke in Jenkins 2.2 for new agents created from the UI (copying existing agents and from the CLI probably still worked).
     * Nobody noticed for almost four years.
     * Recording of the agent creator's userId was removed without the usual plugin detachment process because the issue was undetected for such a long time.
     * See JENKINS-21837 for more details.
```

Reply:
Looks more like a job for the Git history.